src/serializer/ResidualHeapSerializer.js

```diff
@@ -206,18 +205,11 @@ export class ResidualHeapSerializer {
   // TODO: revisit this and fix additional functions to be capable of delaying initializations
   additionalFunctionValueNestedFunctions: Set<FunctionValue>;
   currentAdditionalFunction: void | FunctionValue;
-  functionNames: Map<FunctionValue, string>;
-
-  _getFunctionName(f: FunctionValue): string {
-    let n = this.functionNames.get(f);
-    if (n === undefined) this.functionNames.set(f, (n = this.functionNameGenerator.generate(f.__originalName || "")));
-    return n;
-  }
 
   _getScopeName(s: Scope): string {
-    if (s instanceof Generator) return `#${s.id}`;
+    if (s instanceof Generator) return s.getName();
```

Comment:
Seems like this entire function should just boil down to `s.getName()`, or isn't Flow up to it?

Reply:
Makes sense, let me try.